dir.targets

```diff
@@ -28,6 +28,15 @@
     <TargetFrameworkProfile></TargetFrameworkProfile>
     <!-- We set this property to avoid MSBuild errors regarding not setting TargetFrameworkProfile (see above line) -->
     <PortableNuGetMode>true</PortableNuGetMode>
+
+    <!--Start with NonUnixTests if we aren't on Windows and overwrite with a specific OS, if we know what it is-->
+    <TestFilterCategory Condition="'$(OS)' != 'Windows_NT'">-notrait category=nonunixtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'OSX'">-notrait category=nonosxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Windows_NT'">-notrait category=nonwindowstests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'Linux'">-notrait category=nonlinuxtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'FreeBSD'">-notrait category=nonfreebsdtests</TestFilterCategory>
+    <TestFilterCategory Condition="'$(OS)' == 'NetBSD'">-notrait category=nonnetbsdtests</TestFilterCategory>
+    <TestCommandLine>$(TestCommandLine) $(TestFilterCategory)</TestCommandLine>
```

Comment:
When I last spoke with @weshaggard he mentioned that he usually puts changes like this into CoreFx to vet it first and then moves it over to build tools after.